ACE/ace/Flag_Manip.cpp

```diff
@@ -21,43 +21,90 @@ ACE_BEGIN_VERSIONED_NAMESPACE_DECL
 
 // Flags are file status flags to turn on.
 
+#ifndef ACE_LACKS_FCNTL
+namespace {
+  int set_flags_helper (ACE_HANDLE handle, int flags)
+  {
+    int val = ACE_OS::fcntl (handle, F_GETFL, 0);
+
+    if (val == -1)
+      return -1;
+
+    // Turn on flags.
+    ACE_SET_BITS (val, flags);
+
+    if (ACE_OS::fcntl (handle, F_SETFL, val) == -1)
+      return -1;
+    else
+      return 0;
+  }
+
+  int clr_flags_helper (ACE_HANDLE handle, int flags)
+  {
+    int val = ACE_OS::fcntl (handle, F_GETFL, 0);
+
+    if (val == -1)
+      return -1;
+
+    // Turn flags off.
+    ACE_CLR_BITS (val, flags);
+
+    if (ACE_OS::fcntl (handle, F_SETFL, val) == -1)
+      return -1;
+    else
+      return 0;
+  }
+}
+#endif
+
+// INTEGRITY 11.4.6 simulator does not support ACE_NONBLOCK flag.
+// ACE_USES_GHS_ISIMPPC is defined when running on the simulator.
 int
 ACE::set_flags (ACE_HANDLE handle, int flags)
 {
   ACE_TRACE ("ACE::set_flags");
-#if defined (ACE_LACKS_FCNTL)
+#if defined (ACE_USES_GHS_ISIMPPC)
+  switch (flags)
+    {
+    case ACE_NONBLOCK:
+      ACE_UNUSED_ARG (handle);
```

Comment:
Shouldn't this be a not sup return -1, the higher layer has explicitly requested nonblock which is not supported, currently this requested flag is just silently ignored

Reply:
This is specific to the INTEGRITY simulator which doesn't support non-blocking mode. Returning 0 allows the applications/tests to continue on the simulator and running with blocking mode is acceptable in those programs. That is, it isn't an error indication in this case.